### DISCUSSION TITLE
memory problem

### DISCUSSION THREAD

**Discussion by plumbum082** [deepmodeling/deepks-kit#25] [Q&A] [answered]
Hi, I seemed to run out of memory when I was testing water hexamer using aug-cc-pvqz basis set, it says

_systems/test6_0 failed! error: Unable to allocate 1.03 TiB for an array with shape (142059690906,) and data type float64
starting calculation with OMP threads: 40 and max memory: 320000_

However, the available memory of the machine can not satisfy the need of 1.03TB, I don't know whether it really need so much memory or it is due to other causes. I am not sure what can I do to solve this problem...Could you please help me?
By the way, actually, I don't have the root authority, and I can't use this command "echo 1 > /proc/sys/vm/overcommit_memory"...

**Comment by y1xiaoc** [comment, marked as answer]
Hi, I think this is because the system you are calculating is too large. please comment out the line `incore_anyway: True` in scf_input, mol_args (https://github.com/deepmodeling/deepks-kit/blob/9bbec5232425e54ed7bb8cb25201d0a8f3f50281/examples/water_cluster/args.yaml#L26 for the water example). This is a setting that will be directly passed to pyscf. Please check pyscf's doc for detailed reference. Thanks!

**Reply by plumbum082** [reply]
okay! Thanks a lot for your help!
